assignments/prototype-refactor.js

```diff
@@ -7,3 +7,142 @@ Prototype Refactor
 2. Your goal is to refactor all of this code to use ES6 Classes. The console.log() statements should still return what is expected of them.
 
 */
+
+/*
+  Object oriented design is commonly used in video games.  
+  For this part of the assignment you will be implementing several constructor functions with their correct inheritance heirarchy.
+
+  In this file you will be creating three constructor functions: GameObject, CharacterStats, Humanoid.  
+
+  At the bottom of this file are 3 objects that all end up inheriting from Humanoid.  
+  Use the objects at the bottom of the page to test your constructor functions.
+  
+  Each constructor function has unique properites and methods that are defined in their block comments below:
+*/
+  
+/*
+  === GameObject ===
+  * createdAt
+  * dimensions
+  * destroy() // prototype method -> returns the string: 'Object was removed from the game.'
+*/
+
+  class GameObject{
+      constructor(stats){
+        this.createdAt = stats.createdAt;
+        this.dimensions = stats.dimensions;
+      }
+      destroy(){
+        return `${this.name} was removed from the game.`;
```

Comment:
You might want to set up a default for your `this.name` here since `GameObject` doesn't have a `name` property-- what if you were to destroy a non-character object? It'd return `undefined was removed from the game.` Here's one way of doing that using a [ternary](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Conditional_Operator), which is the equivalent of an `if ... else ...`:
```javascript
destroy() {
    const name = this.name ? this.name : 'Object';
    return `${name} was removed from the game.`;
}
```